controller/instance_manager_controller.go

```diff
@@ -920,52 +922,52 @@ func (imc *InstanceManagerController) deleteInstanceManagerPDB(im *longhorn.Inst
 	return nil
 }
 
-func (imc *InstanceManagerController) canDeleteInstanceManagerPDB(im *longhorn.InstanceManager) (bool, error) {
+func (imc *InstanceManagerController) canDeleteInstanceManagerPDB(im *longhorn.InstanceManager) (bool, string, error) {
 	// If there is no engine instance process inside the engine instance manager,
 	// it means that all volumes are detached.
 	// We can delete the PodDisruptionBudget for the engine instance manager.
 	if im.Spec.Type == longhorn.InstanceManagerTypeEngine {
 		if len(im.Status.InstanceEngines)+len(im.Status.Instances) == 0 { // nolint: staticcheck
-			return true, nil
+			return true, "", nil
 		}
-		return false, nil
+		return false, "some instances are still running", nil
```

Comment:
If we are going to add this extra logging, I think we should take it a step further and log exactly which instances are running. It makes this function a bit more complex, but could be very useful. WDYT?